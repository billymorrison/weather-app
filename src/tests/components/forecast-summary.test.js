import React from 'react'
import { render } from '@testing-library/react'
import ForecastSummary from '../../components/forecastsummary'

const mockFunction = () => {
    return
}

describe('Forecast Summary', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <ForecastSummary
                date={1606129635000}
                temperature={68494}
                description='testDesc'
                icon='800'
                onSelect={mockFunction}
            />
        )

        expect(asFragment()).toMatchSnapshot();
    })

    it('renders a number of props into a weather summary', () => {
        const { getByText, getByTestId } = render(
            <ForecastSummary
                date={1606129635000}
                temperature={9}
                description='Clear'
                icon='800'
                onSelect={mockFunction}
            />
        )

        expect(getByText('Monday Nov 23rd')).toBeTruthy();
        expect(getByText('Monday Nov 23rd')).toHaveClass('date');
        expect(getByTestId('temperature-id')).toHaveClass('max-temperature');
        expect(getByText('Clear')).toHaveClass('description');
    })
})