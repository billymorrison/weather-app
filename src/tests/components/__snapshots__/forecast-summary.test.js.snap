// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Summary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <span
      class="date"
      data-testid="date-id"
    >
      Monday Nov 23rd
    </span>
    <i
      class="wi wi-day-sunny"
      data-testid="icon-image-id"
      name="owm"
    />
    <span
      class="max-temperature"
      data-testid="temperature-id"
    >
      68494â„ƒ
    </span>
    <span
      class="description"
      data-testid="description-id"
    >
      testDesc
    </span>
    <button>
      More Details
    </button>
  </div>
</DocumentFragment>
`;
